File: 1574-maximum-product-of-two-elements-in-an-array/maximum-product-of-two-elements-in-an-array.java
class Solution {
    public int maxProduct(int[] nums) {
        int m1 = nums[0];
        int m2 = nums[1];
        int n = nums.length;
        for(int i = 1 ; i < n ; i++){
            if(nums[i] > m1){
                m2 = m1;
                m1 = nums[i];
            }else if(nums[i] > m2){
                m2 = nums[i];
            }
        }
        return (m1 - 1) * (m2 - 1);
    }
}

File: 1574-maximum-product-of-two-elements-in-an-array/maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int prod = 0;
        for(int i = 0 ; i < nums.size() ; i++){
            for(int j = i+1 ; j < nums.size() ; j++){
                prod = max((nums[i]-1) * (nums[j]-1), prod);
            }
        }
        return prod;
    }
};